streams/src/test/java/org/apache/kafka/streams/state/internals/CachingInMemoryKeyValueStoreTest.java

```diff
@@ -221,17 +223,44 @@ public void shouldPutGetToFromCache() {
     }
 
     @Test
-    public void shouldMatchPositionAfterPut() {
+    public void shouldMatchPositionAfterPutWithFlushListener() {
+        store.setFlushListener(record -> {}, false);
+        shouldMatchPositionAfterPut();
+    }
+
+    @Test
+    public void shouldMatchPositionAfterPutWithoutFlushListener() {
+        store.setFlushListener(null, false);
+        shouldMatchPositionAfterPut();
+    }
```

Comment:
If you look at the caching stores above, you'll see they have separate branches, depending on whether there's a flush listener defined, so we need to test both branches. Ditto with the other tests.